### PR TITLE
Blog app - integration specs for views

### PR DESCRIPTION
**Pull request by olawale-o**
# In this PR, we did the following:
 - Use Capybara to write test
- Login page:

   - I can see the username and password inputs and the "Submit" button.
   - When I click the submit button without filling in the username and the password, I get a detailed error.
   - When I click the submit button after filling in the username and the password with incorrect data, I get a detailed error.
   - When I click the submit button after filling in the username and the password with correct data, I am redirected to the root page.

- User index page:

  - I can see the username of all other users.
   - I can see the profile picture for each user.
   - I can see the number of posts each user has written.
   - When I click on a user, I am redirected to that user's show page.

- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

### CODE REVIEW

File: spec/features/login/second_login_form_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      click_button 'Log in'
+      expect(page.has_content?('Invalid Email or password.')).to be_truthy
+    end
+
+    scenario 'with valid email and password' do
+      within 'form' do
+        fill_in 'user[email]', with: user.email
+        fill_in 'user[password]', with: user.password
+      end
+      click_button 'Log in'
```
Comment (Emmyn5600) [inline spec/features/login/second_login_form_spec.rb:27]: - [ ] Optional suggestion. You can also create the `third_login_form_spec` with the following implementation

```
 describe 'the signin process', type: :feature do
     before :each do
         User.create(email: 'user1@gmail.com', password: 'password', name: 'User1')
     end

     it 'signs @user in' do
         visit '/users/sign_in'
         fill_in 'Email', with: 'user1@gmail.com'
         fill_in 'Password', with: 'password'
         click_button 'Log in'
         expect(current_path).to eq(root_path)
         expect(page).to have_text('Signed in successfully.')
     end

     it 'does not sign in with wrong info' do
         visit '/users/sign_in'
         fill_in 'Email', with: 'user1@gmail.com'
         fill_in 'Password', with: 'mypassword'
         click_button 'Log in'
         expect(page).to have_text('Invalid Email or password.')
     end
 end
```

**Review by Emmyn5600 — APPROVED** [review]
Hi @olawale-o  and @PolinaStamenova 

Good job👏🏼
- 55 tests are passing successfully.

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the project requirements.

 #### Optional suggestions
 -  I would recommend you to use  `binding.pry and :chrome` to verify that your tests are working, this will make your life be much easier if you do the above two when writing capybara tests. But you're not supposed to push that, those are for development only.
  
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: spec/features/login/second_login_form_spec.rb (hunk 1 above)
Comment (olawale-o) [reply spec/features/login/second_login_form_spec.rb:27]: Hi @Emmyn5600 
Thank you for this great suggestion

**Comment by olawale-o** [comment]
Hi @Emmyn5600 
Thank you for your time and effort on the review

**Comment by Emmyn5600** [comment]
@olawale-o, If you have time you can read more about `binding.pry and :chrome ` if you need to know the purpose of debugging while you are writing tests to check if there are running successfully.

**Comment by olawale-o** [comment]
> @olawale-o, If you have time you can read more about `binding.pry and :chrome ` if you need to know the purpose of debugging while you are writing tests to check if it is running successfully.

Thank you very much for pointing me to these helpful resources. I really appreciate it

**Pull request merged**